pkg/credentialprovider/plugin/examples/ecr/main.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// TODO: this is just a reference implementation, this plugin should be moved
```

Comment:
@nckturner this is just a reference implementation for ECR, can you review this and provide feedback on whether it meets all the requirements based on the existing provider? At some point this should be moved to kubernetes/cloud-provider-aws.

Reply:
btw, I've tested the ECR plugin using this reference implementation and it works

Reply:
Will review this tonight, sorry for the delay.